multiqc/modules/prokka/prokka.py

```diff
@@ -0,0 +1,153 @@
+        
+        # Add Prokka's annotation stats to the general stats table
+        headers = OrderedDict()
+        headers['organism'] = {
+            'title': 'Organism',
+            'description': 'Organism',
+        }
+        #headers['contigs'] = {
+        #    'title': 'Contigs',
+        #    'description': 'Number of contigs',
+        #    "min": 0,
+        #}
+        headers['bases'] = {
+            'title': 'Bases',
+            'description': 'Number of bases',
+            "min": 0,
+            'format': '{:i}%'
+        }
+        headers['CDS'] = {
+            'title': 'CDS',
+            'description': 'Number of CDS',
+            "min": 0,
+            'format': '{:i}%'
+        }
+        headers['tRNA'] = {
+            'title': 'tRNA',
+            'description': 'Number of tRNA',
+            "min": 0,
+            'format': '{:i}%'
+        }
+        headers['rRNA'] = {
+            'title': 'rRNA',
+            'description': 'Number of rRNA',
+            "min": 0,
+            'format': '{:i}%'
+        }
+        headers['tmRNA'] = {
+            'title': 'tmRNA',
+            'description': 'Number of tmRNA',
+            "min": 0,
+            'format': '{:i}%'
+        }
+        headers['sig_peptide'] = {
+            'title': 'sig_peptide',
+            'description': 'Number of sig_peptide',
+            "min": 0,
+            'format': '{:i}%'
+        }
+        self.general_stats_addcols(self.prokka, headers)
+        
+        # Make summary barplot
+        # The stacked barplot is not really useful for Prokka
+        self.intro += self.prokka_barplot()
+        
+    def parse_prokka(self, f):
+        """ Parse prokka txt summary files. 
+        
+        Prokka summary files are difficult to identify as there are practically
+        no distinct prokka identifiers in the filenames or file contents. This
+        parser makes an attempt using the first three lines, expected to contain
+        organism, contigs, and bases statistics. 
+        """
+
+        s_name = None
+
+        # Look at the first three lines, they are always the same
+        first_line = f['f'].readline()
+        contigs_line = f['f'].readline()
+        bases_line = f['f'].readline()
+        # If any of these fail, it's probably not a prokka summary file
+        if not all((first_line.startswith("organism:"), 
+                   contigs_line.startswith("contigs:"),
+                   bases_line.startswith("bases:"))):
+            return
+        log.debug("%s appears to be a prokka summary file.", f['f'].name)
+
+        # Get organism and sample name from the first line
+        organism = " ".join(first_line.strip().split(":", maxsplit=1)[1].split()[:2])
+        s_name = " ".join(first_line.split()[3:])
+        self.prokka[s_name] = dict()
+        self.prokka[s_name]['organism'] = organism
+        log.debug("Found sample name '%s' for organism '%s'", s_name, organism)
```

Comment:
This should be in the exported data file? Probably nice to skip it to not to make the logfile too big..